Cleanup various "nil vs empty" confusions in serializaiton of some ObjC types

beidson opened the pull request:
#### 26b41e6069aaf52423f7f73664cf8c29aef70bd8
<pre>
Cleanup various &quot;nil vs empty&quot; confusions in serializaiton of some ObjC types
<a href="https://rdar.apple.com/128027011">rdar://128027011</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=274342">https://bugs.webkit.org/show_bug.cgi?id=274342</a>

Reviewed by Alex Christensen.

Similar to what I fixed over in <a href="https://commits.webkit.org/277627@main">https://commits.webkit.org/277627@main</a>, various other ObjC serializers
have some &quot;nil vs empty&quot; confusion that can lead to various wonky side effects.

Let&apos;s support the different between nil and empty wherever we can spot it.

* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.h:
(WebKit::CoreIPCPKContact::CoreIPCPKContact):
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.mm:
(WebKit::CoreIPCPKContact::CoreIPCPKContact):
(WebKit::CoreIPCPKContact::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:

* Source/WebKit/Shared/Cocoa/CoreIPCPersonNameComponents.mm:
(WebKit::CoreIPCPersonNameComponents::toID const):

* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(TEST(IPCSerialization, Basic)):

Canonical link: <a href="https://commits.webkit.org/278966@main">https://commits.webkit.org/278966@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6f04da31d14c216baa0c825e3702c139e09f899d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52049 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/31374 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/4458 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/55322 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/2761 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/54351 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/37756 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/2467 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42365 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/1759 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54143 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28993 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44894 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23434 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26276 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2164 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/931 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48173 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/2314 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56919 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27164 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/2311 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49758 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/28402 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45014 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48976 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11402 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/29305 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28142 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/Cocoa/CoreIPCPassKit.mm:
```diff
@@ -33,26 +33,33 @@
     , m_supplementarySublocality(contact.supplementarySubLocality)
 ALLOW_DEPRECATED_DECLARATIONS_END
 {
+    if (contact.name)
```
These could be still in the initializers with ternary operators

beidson replied:
That was the first attempt, but ternary + std::optional == a bad time.

beidson committed:
Cleanup various "nil vs empty" confusions in serializaiton of some ObjC types
rdar://128027011
https://bugs.webkit.org/show_bug.cgi?id=274342

Reviewed by Alex Christensen.

Similar to what I fixed over in https://commits.webkit.org/277627@main, various other ObjC serializers
have some "nil vs empty" confusion that can lead to various wonky side effects.

Let's support the different between nil and empty wherever we can spot it.

* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.h:
(WebKit::CoreIPCPKContact::CoreIPCPKContact):
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.mm:
(WebKit::CoreIPCPKContact::CoreIPCPKContact):
(WebKit::CoreIPCPKContact::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:

* Source/WebKit/Shared/Cocoa/CoreIPCPersonNameComponents.mm:
(WebKit::CoreIPCPersonNameComponents::toID const):

* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(TEST(IPCSerialization, Basic)):

Canonical link: https://commits.webkit.org/278966@main
